internal/util/parse.go

```diff
@@ -110,3 +110,12 @@ func ParseBool(b string) *bool {
 	}
 	return &truth
 }
+
+// ReadHexFromFile reads a file and attempts to parse a uint64 from a hexadecimal format 0xXX.
+func ReadHexFromFile(path string) (uint64, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return 0, err
+	}
+	return strconv.ParseUint(strings.TrimSpace(string(data[2:])), 16, 64)
```

Comment:
This should validate that the string starts with `0x` before trying to parse the value.

Reply:
Updated